### PR TITLE
Run tests with utf-8 encoding

### PR DESCRIPTION
**Pull request by jdemeyer**

### CODE REVIEW

File: autogen/generator.py [hunk 1]
```diff
@@ -320,11 +320,16 @@ def __call__(self):
         D = sorted(D.values(), key=lambda d: d['function'])
         sys.stdout.write("Generating PARI functions:")
 
-        self.gen_file = open(self.gen_filename + '.tmp', 'w')
+        # Stupid Python 3 forces us to specify an encoding
+        if "encoding" in open.__doc__:
+            kwds = dict(mode="wt", encoding="utf-8")
+        else:
+            kwds = dict(mode="wt")
```
Comment (embray) [inline autogen/generator.py]: The "cross-python" compatible way to write this:

```
import io
f = io.open(filename, 'w', encoding='utf-8')
```

if you want to specify UTF-8.  On Python 3, `builtins.open` *is* `io.open`.  On Python 2 it is not (`open` is the legacy built-in from before the I/O refactoring, which in Python 2 is fully encapsulated in the `io` module).

For Python2/3 compatibility w.r.t. I/O it's almost always best to use the `io` module explicitly.

Comment (jdemeyer) [inline autogen/generator.py]: Right. For this to work, I need to do everything in `unicode` though. Right now, I do an explicit encoding `unicode` -> `str` on Python 2: https://github.com/defeo/cypari2/blob/master/autogen/generator.py#L289

So I would need to delete that and then use `from __future__ import unicode_literals` everywhere.

Comment (jdemeyer) [inline autogen/generator.py]: Aargh...
```
**********************************************************************
File "/usr/local/src/sage-config/local/src/cypari2/autogen/parser.py", line 104, in autogen.parser.parse_prototype
Failed example:
    parse_prototype("lp", "foo()", ["TEST"])
Differences (ndiff with -expected +actual):
    - (['TEST', prec precision=0], long)
    + ([u'TEST', prec precision=0], long)
    ?   +
**********************************************************************
```

Comment (embray) [inline autogen/generator.py]: One thing we really should have proposed for Python when trying to improve the 2/3 compatibility situation is to add a doctest directive to ignore unicode literal markers, b.c. I've had to reimplement that multiple times...  I think in general doctests just aren't used as extensively in much of the Python community, but still.

Also I don't think it's necessary to put `unicode_literals` absolutely everywhere but I haven't done a careful analysis of where the bytes/unicode boundaries are in the autogen code :/

**Comment by jdemeyer** [comment]
> Also I don't think it's necessary to put unicode_literals absolutely everywhere but I haven't done a careful analysis of where the bytes/unicode boundaries are in the autogen code :/

All those files do contain strings which end up in the auto-generated code, so I think it is needed everywhere.

**Pull request merged**